src/testrunner.js

```diff
@@ -0,0 +1,349 @@
+  let adherencePercentage = (testVerdictCounter.pass/
+      (testVerdictCounter.pass + testVerdictCounter.fail))*100;
+  adherencePercentage = Math.round(adherencePercentage * 100) / 100;
+  logger.info('adherence percentage: '.grey + `${adherencePercentage}`.cyan);
+}
+
+/**
+ * displays the time taken for the execution of testsuite
+ * @param {number} startTime time at which execution of testsuite started
+ * @param {number} endTime time at which execution of testsuite ended
+ */
+function displayTimeTaken(startTime, endTime) {
+  let millisecond = endTime - startTime;
+  millisecond = Math.round(millisecond * 100) / 100;
+  logger.info('time taken: '.grey + `${millisecond} ms`);
+}
+
+/**
+ * displays test results of each test case with test verdicts, test case
+ * details and other details like errors if any.
+ * @param {object} testResults
+ */
+function displayTestResults(testResults) {
+  testResults.forEach(function(testResult) {
+    const {
+      testCase,
+      testVerdict,
+      skipValidation,
+      statusCodes,
+      errors} = testResult;
+
+    /*
+      Filter out unnecessary data from the testcase ,
+      in order to present only the necessary details to the user.
+    */
+    delete testCase.data;
+    delete testCase.testForRequestBody;
+
+    const verdictLog =
+        ((testVerdict.final === 'pass') ? 'PASS'.green : 'FAIL'.red) + ' ' +
+        JSON.stringify(testCase).grey.bold;
+    const statusCodeLog = 'status codes: ' +
+        `[received: ${statusCodes.received}, ` +
+        `expected: ${JSON.stringify(statusCodes.expected)} ]`;
+    const skipLog = 'skipped' +
+        (skipValidation.responseBody ? ' [response body]': '') +
+        (skipValidation.responseHeaders ? ' [response header]': '') +
+        ' validation.';
+
+    logger.info(verdictLog);
+    logger.verbose(statusCodeLog);
+    if (skipValidation.responseBody || skipValidation.responseHeaders) {
+      logger.verbose(skipLog);
+    }
+    logger.verbose('\n');
+
+    if (testVerdict.final === 'pass') {
+      testVerdictCounter.pass ++;
+    } else {
+      testVerdictCounter.fail ++;
+      if (testVerdict.initial === 'pass') {
+        if (errors.responseBody.length) {
+          logger.error('Response Body Errors: '.yellow.bold);
+          logger.error(`${JSON.stringify(errors.responseBody)}`.magenta);
+        }
+        if (errors.responseHeaders.length) {
+          logger.error('Response Header Errors: '.yellow.bold);
+          logger.error(`${JSON.stringify(errors.responseHeaders)}`.magenta);
+        }
+      }
+    }
+  });
+}
+
+/**
+ * executes testcases and generates test results for a particular api endpoint
+ * @param {array<object>} testCases
+ * @param {array<string>} expectedStatusCodes
+ * @param {object} apiTestSuite  testsuites of an api endpoint
+ * @param {object} oasDoc  oas 3.0 document
+ * @param {object} axiosConfig contains configs of an axios request
+ */
+async function runTestCase(testCases, expectedStatusCodes,
```

Comment:
Function is too big, can be broken down further

Reply:
Made function more modular.